Content.Server/Storage/EntitySystems/StorageSystem.cs

```diff
@@ -34,6 +38,30 @@ public override void Initialize()
             SubscribeLocalEvent<EntityStorageComponent, GetInteractionVerbsEvent>(AddToggleOpenVerb);
             SubscribeLocalEvent<ServerStorageComponent, GetActivationVerbsEvent>(AddOpenUiVerb);
             SubscribeLocalEvent<EntityStorageComponent, RelayMovementEntityEvent>(OnRelayMovement);
+            SubscribeLocalEvent<EntityStorageComponent, ContainerGettingInsertedAttemptEvent>(OnInsertAttempt);
+        }
+
+        private void OnInsertAttempt(EntityUid uid, EntityStorageComponent component, ContainerGettingInsertedAttemptEvent args)
+        {
+            // This entity-storage is being inserted into a container. In general this is fine (e.g., pizza box in a
+            // locker). But maybe, just maybe, this is a player somehow being bugged into a backpack.
+
+            // Firstly, check if the container is a backpack/storage entity?
+            if (!HasComp<SharedStorageComponent>(args.Container.Owner))
+                return;
+
+            // Ok, so we will check if any of the entities we contain is a mob, and if it is, we cancel this attempt. In
+            // general, nesting entity-storage inside of a storage is rare. So just checking every contained entity
+            // every time should be fine. The only instances I can think of is pizza boxes and body-bags.
+
+            foreach (var ent in component.Contents.ContainedEntities)
+            {
+                if (HasComp<MobStateComponent>(ent) || HasComp<SharedBodyComponent>(ent))
+                {
+                    args.Cancel();
+                    return;
+                }
+            }
```

Comment:
I'm pretty suss of this and feel sizes should be used instead.

e.g. If a mob is inside a livestock crate and you want to insert that crate into a bigger crate that shouldn't necessarily be blocked as it feels a bit hardcode-y.

Reply:
Just to be clear, this shouldn't block that example this is specifically for `SharedStorageComponent` (storage with a UI like backpacks &  boxes), not entity-storage like lockers, creates, etc.

But even with that caveat I also don't like having to go out of my way to specifically preventing mobs being inserted. Heck, I couldn't even decide between whether `MobStateComponent` or `SharedBodyComponent` should define a mob.

I guess a better solution would be to just generalize the requirement to:
When inserting an entity into storage, check if that entity has any containers. If it does, only allow that entity if every contained entity would also be individually allowed in.

That would probably be a better and more general way of preventing people-in-backpacks shenanigans.